packages/sol/src/verification.ts

```diff
@@ -0,0 +1,389 @@
+import {
+  COUPON_REGEX,
+  DISCOUNT_MAX_VALUE,
+  DISCOUNT_MIN_VALUE,
+  EMAIL_REGEX,
+  INSTAPAY_REGEX,
+  INTERVIEW_MAX_LEVEL,
+  INTERVIEW_MIN_LEVEL,
+  LETTERS_ONLY_REGEX,
+  MINUTES_IN_WEEK,
+  NUMBERS_ONLY_REGEX,
+  PHONE_NUMBER_REGEX,
+  RATING_MAX_VALUE,
+  RATING_MIN_VALUE,
+  REGEXFORHTML,
+  RULE_DURATION_MAX,
+  USER_NAME_REGEX,
+} from "@/constants";
+import { getTextFromHTMLString } from "./utils";
+import { sanitizeMessage } from "./chat";
+
+export function isValidEmail(email: string): FieldError.InvalidEmail | true {
+  if (!EMAIL_REGEX.test(email)) return FieldError.InvalidEmail;
+  return true;
+}
+
+export function isValidPassword(
+  password: string
+):
+  | FieldError.PasswordTooShort
+  | FieldError.PasswordMissingLetters
+  | FieldError.PasswordMissingNumbers
+  | true {
+  if (!NUMBERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingNumbers;
+  if (!LETTERS_ONLY_REGEX.test(password))
+    return FieldError.PasswordMissingLetters;
+  if (password.length < 8) return FieldError.PasswordTooShort;
+  return true;
+}
+
+export function isValidUserName(
+  userName: string
+): FieldError.InvalidUserName | true {
+  if (!USER_NAME_REGEX.test(userName)) return FieldError.InvalidUserName;
+  return true;
+}
+
+export function isValidUserBirthYear(
+  userBirthYear: number
+): FieldError.TutorTooOld | FieldError.TutorTooYoung | true {
+  const currentYear = new Date().getFullYear();
+  const minYear = currentYear - 60;
+  const maxYear = currentYear - 10;
+
+  if (userBirthYear < minYear) return FieldError.TutorTooOld;
+  if (userBirthYear > maxYear) return FieldError.TutorTooYoung;
+  return true;
+}
+
+export function isValidTutorBio(
+  tutorBio: string
+):
+  | FieldError.TutorBioInvalidHTML
+  | FieldError.TutorBioEmpty
+  | FieldError.TutorBioTooLong
+  | true {
+  const tutorBioText = getTextFromHTMLString(tutorBio);
+  console.log("text", tutorBioText);
+  if (!tutorBioText.length) return FieldError.TutorBioEmpty;
+  if (!REGEXFORHTML.test(tutorBio)) return FieldError.TutorBioInvalidHTML;
+  if (tutorBioText.length > 60) return FieldError.TutorBioTooLong;
+  return true;
+}
+
+export function isValidTutorAbout(
+  tutorAbout: string
+):
+  | FieldError.TutorAboutInvalidHTML
+  | FieldError.TutorAboutEmpty
+  | FieldError.TutorAboutTooLong
+  | true {
+  const tutorAboutText = getTextFromHTMLString(tutorAbout);
+
+  if (!tutorAboutText.length) return FieldError.TutorAboutEmpty;
+  if (!REGEXFORHTML.test(tutorAbout)) return FieldError.TutorAboutInvalidHTML;
+  if (tutorAboutText.length > 1000) return FieldError.TutorAboutTooLong;
+  return true;
+}
+
+export function isValidTutorNotice(
+  tutorNotice: number
+): FieldError.TutorNoticeTooLong | FieldError.TutorNoticeMinus | true {
+  if (tutorNotice < 0) return FieldError.TutorNoticeMinus;
+  if (tutorNotice > 24 * 60) return FieldError.TutorNoticeTooLong;
+  return true;
+}
+
+export function isValidRuleTitle(
+  ruleTitle: string
```

Comment:
Short names are preferred. Try to be something in the middle. Don't be very verbose and very vague. 
 
```suggestion
  title: string
```